answerking_app/tests/BaseTestClass.py

```diff
@@ -0,0 +1,184 @@
+    }
+    expected_duplicated_name_error: DetailError = {
+        "detail": "This name already exists",
+        "status": 400,
+        "title": "A server error occurred.",
+        "type": "http://testserver/problems/error/",
+    }
+    expected_invalid_status = {
+        "detail": "Object was not Found",
+        "errors": ["Status matching query does not exist."],
+        "status": 404,
+        "title": "Resource not found",
+        "type": "http://testserver/problems/error/",
+    }
+
+    post_mock_item: ItemType = {
+        "name": "Whopper",
+        "price": "1.50",
+        "description": "desc",
+        "stock": 100,
+        "calories": 100,
+    }
+
+    invalid_json_data: str = '{"invalid": }'
+
+    def setUp(self):
+        self.test_item_1: Item = Item.objects.create(
+            name="Burger",
+            price=1.20,
+            description="desc",
+            stock=100,
+            calories=100,
+        )
+        self.test_item_2: Item = Item.objects.create(
+            name="Coke",
+            price=1.50,
+            description="desc",
+            stock=100,
+            calories=100,
+        )
+        self.test_item_3: Item = Item.objects.create(
+            name="Chips",
+            price=1.50,
+            description="desc",
+            stock=100,
+            calories=100,
+        )
+
+        self.test_cat_1: Category = Category.objects.create(name="Burgers")
+        self.test_cat_2: Category = Category.objects.create(name="Sides")
+
+        self.test_cat_1.items.add(self.test_item_1)
+        self.test_cat_1.items.add(self.test_item_2)
+        self.status_pending: Status = Status.objects.create(status="Pending")
+        self.status_complete: Status = Status.objects.create(
+            status="Completed"
+        )
+
+        self.test_order_empty: Order = Order.objects.create(
+            address="123A3 Street, Leeds, LS17PP",
+            status=self.status_pending,
+        )
+
+        self.test_order_1: Order = Order.objects.create(
+            address="123 Street, Leeds, LS73PP",
+            status=self.status_pending,
+        )
+
+        self.test_order_2: Order = Order.objects.create(
+            address="456 Test Lane, Bradford, BD30PA",
+            status=self.status_pending,
+        )
+
+        self.test_order_1.order_items.add(
+            self.test_item_1,
+            through_defaults={
+                "quantity": 2,
+                "sub_total": f"{self.test_item_1.price * 2:.2f}",
+            },
+        )
+
+        self.test_order_1.order_items.add(
+            self.test_item_2,
+            through_defaults={
+                "quantity": 1,
+                "sub_total": f"{self.test_item_2.price:.2f}",
+            },
+        )
+        self.test_order_1.calculate_total()
+
+    def tearDown(self):
+        Item.objects.all().delete()
+        Category.objects.all().delete()
+        Status.objects.all().delete()
```

Comment:
need to tear down the orders that you have created too

Reply:
yep, missed that 👍